cns/restserver/util.go

```diff
@@ -350,11 +359,145 @@ func (service *HTTPRestService) removeToBeDeletedIPStateUntransacted(
 	return 0, ""
 }
 
+func (service *HTTPRestService) getNetworkContainersResponse(
+	req cns.GetNetworkContainerRequest,
+) (getNetworkContainersResponse []cns.GetNetworkContainerResponse) {
+	var (
+		networkContainerID          string
+		getNetworkContainerResponse cns.GetNetworkContainerResponse
+		exists                      bool
+		waitingForUpdate            bool
+	)
+
+	service.Lock()
+	defer service.Unlock()
```

Comment:
It's never a good idea to have these methods exposed on a type unless the type is a synchronization primitive (like `sync.Cond`, a condition variable). Even if these weren't exposed here, you should use the underlying mutex directly.

Reply:
Can we discuss more about this

Reply:
As discussed, if we change this, we will change everywhere and it might be not high priority now,